File: java/src/org/gessinger/gepard/xmp/EmitterWithBody.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;
import java.util.* ;
public class EmitterWithBody
{
  static public void main ( String[] args )
  {
    Util.argsToProperties ( args ) ;
    try
    {
      EmitterWithBody j = new EmitterWithBody() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
  EmitterWithBody()
  throws Exception
  {
    Client client = Client.getInstance() ;
    String name = Util.getProperty ( "name", "ALARM" ) ;

    Event e = new Event ( name ) ;
    e.putValue ( "BINARY", new byte[] { 11, 12, 13 } ) ;
    e.putValue ( "DATE", new Date() ) ;
    client.emit ( e ) ;
    client.close() ;
  }
}

File: python/xmp/EmitterWithBody.py
#!/usr/bin/env python

import os, sys, datetime
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )
from gepard import Client, Event

client = Client.getInstance()
e = Event ( "ALARM" )
e.putValue ( "STRING", "TEXT" )
e.putValue ( "BINARY", bytearray ( [ 64, 65, 66, 67 ] ) )
e.putValue ( "DATE", datetime.datetime.now() )
client.emit ( e )
client.close()
